src/pages/UDT/index.tsx

```diff
@@ -1,59 +1,52 @@
-import { Link, useHistory, useLocation, useParams } from 'react-router-dom'
+import { useHistory, useParams } from 'react-router-dom'
 import { useTranslation } from 'react-i18next'
 import { useQuery } from '@tanstack/react-query'
-import { Popover } from 'antd'
 import { FC } from 'react'
 import Content from '../../components/Content'
 import { UDTContentPanel, UDTTransactionTitlePanel } from './styled'
 import UDTComp, { UDTOverviewCard } from './UDTComp'
-import { useIsMobile, usePaginationParamsInPage } from '../../hooks'
+import { usePaginationParamsInPage, useSearchParams } from '../../hooks'
 import Filter from '../../components/Search/Filter'
 import { localeNumberString } from '../../utils/number'
 import { explorerService } from '../../services/ExplorerService'
 import { deprecatedAddrToNewAddr } from '../../utils/util'
 import { QueryResult } from '../../components/QueryResult'
 import { defaultUDTInfo } from './state'
-import { ReactComponent as FilterIcon } from '../../assets/filter_icon.svg'
-import { ReactComponent as SelectedCheckIcon } from '../../assets/selected_check_icon.svg'
 import styles from './styles.module.scss'
 import { Cell } from '../../models/Cell'
-
-enum TransactionType {
-  Mint = 'mint',
-  Transfer = 'normal',
-  Burn = 'destruction',
-}
+import { assert } from '../../utils/error'
 
 export const UDT: FC<{ isInscription?: boolean }> = ({ isInscription }) => {
   const { t } = useTranslation()
-  const isMobile = useIsMobile()
   const { push } = useHistory()
-  const { search } = useLocation()
+  // The typeHash here could be either udtTypeHash or omigaInscriptionInfoTypeHash.
   const { hash: typeHash } = useParams<{ hash: string }>()
   const { currentPage, pageSize: _pageSize, setPage } = usePaginationParamsInPage()
 
-  const query = new URLSearchParams(search)
-  const filter = query.get('filter')
-  const type = query.get('type')
+  const { filter, view_original } = useSearchParams('filter', 'view_original')
+  const isViewOriginal = view_original === '1'
```

Comment:
Why use a number as `1` instead of a boolean as `true`

Reply:
I believe that `0` or `1` representing boolean values in search params is a common convention, while `false` or `true` are less common.

Reply:
> I believe that `0` or `1` representing boolean values in search params is a common convention, while `false` or `true` are less common.

I would suggest using a semantic field, e.g. `view=original`, to make it extensible. `Omiga` has 2 views, `latest` and `original`, but other inscriptions may have more stages. I'm not sure if there will be more stages, but it's easy to have more views with `view=xxx`

Reply:
Makes sense, updated.